pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -93,10 +107,30 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
-		err = client.SetToken(path.Join(instance.Spec.TenantRef.Name, instance.Name, "steward"), token, reqLogger)
+		err = vault.SetToken(secretPath, token)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		vault.SetDeletePaths([]string{secretPath})
```

Comment:
Could we add the functionality of `SetDeletePaths` to `SetToken` so we don't have to call it explicitly?

Reply:
I opted for this solution because `SetToken` expects data and will also create the token.

Reply:
But the `SetToken` call already has all the necessary info, no?
So it would just need to additionally save the secretPath in a map on the vault client instance. Or in other words, every time you call `SetToken` for a path, you also have to call `SetDeletePaths` with the same path which results in a rather bad experience.

Reply:
I redefined the vault interface and implemented the deletion logic changes we discussed.